api/transport/router.go

```diff
@@ -59,12 +60,30 @@ func (p Procedure) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	return enc.AddObject("handler", p.HandlerSpec)
 }
 
-// Less orders procedures lexicographically on (Service, Name).
-func (p Procedure) Less(other Procedure) bool {
-	if p.Service == other.Service {
-		return p.Name < other.Name
+// Less orders procedures lexicographically on (Service, Name, Encoding).
+func (p Procedure) Less(o Procedure) bool {
+	if p.Service != o.Service {
+		return p.Service < o.Service
 	}
-	return p.Service < other.Service
+	if p.Name != o.Name {
+		return p.Name < o.Name
+	}
+	return p.Encoding < o.Encoding
+}
+
+// Procedures is a sortable slice of procedures.
+type Procedures []Procedure
```

Comment:
Adding this to the public API is not needed. This is used in only one place. Unnecessary added API surface.